moto/ec2/models.py

```diff
@@ -209,7 +212,8 @@ def __init__(self, ec2_backend, subnet, private_ip_address, device_index=0,
         self.instance = None
         self.attachment_id = None
 
-        self.public_ip = None
+        self.vm_id = vm_id
+        self._public_ip = None
```

Comment:
Why did you change the member name? It seems not a good idea.

Reply:
See property method public_id below.